packages/contracts/contracts/escrow/CollateralEscrowFactory.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-Licence-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+// Contracts
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+import "./CollateralEscrowV1.sol";
+
+contract CollateralEscrowFactory is UpgradeableBeacon {
+
+    address private _collateralEscrowBeacon;
+    mapping(uint256 => address) public _escrows; // bidIds -> collateralEscrow
+
+    /* Events */
+    event CollateralEscrowDeployed(uint256 _bidId, address collateralEscrow);
+
+    /**
+     * @notice Initializes the escrow factory.
+     * @param _factoryBeacon The address of the factory implementaion.
+     * @param _collateralEscrowBeacon The address of the escrow implementation.
+     */
+    constructor(
+        address _factoryBeacon,
+        address _collateralEscrowBeacon
+    ) UpgradeableBeacon(_factoryBeacon) {
+        _collateralEscrowBeacon = _collateralEscrowBeacon;
+    }
+
+    /**
+     * @notice Deploys a new collateral escrow.
+     * @param _bidId The associated bidId of the collateral escrow.
+     */
+    function deployCollateralEscrow(uint256 _bidId)
+        external
+        returns(address)
+    {
+        BeaconProxy proxy_ = new BeaconProxy(
+            _collateralEscrowBeacon,
+            abi.encodeWithSelector(CollateralEscrowV1.initialize.selector, _bidId)
+        );
+        _escrows[_bidId] = address(proxy_);
```

Comment:
doesnt this mean that anyone can overwrite anyone elses ?

Reply:
removed the escrow factory and replaced with a library in https://github.com/teller-protocol/teller-protocol-v2/pull/6